larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h

```diff
@@ -0,0 +1,393 @@
+/**
+ *  @file   larpandoracontent/LArObjects/LArDiscreteProbabilityVector.h
+ *
+ *  @brief  Header file for the lar discrete probability vector class
+ *
+ *  $Log: $
+ */
+#ifndef LAR_DISCRETE_PROBABILITY_VECTOR_H
+#define LAR_DISCRETE_PROBABILITY_VECTOR_H 1
+
+#include "Pandora/StatusCodes.h"
+
+#include <limits>
+#include <random>
+#include <utility>
+#include <vector>
+
+namespace lar_content
+{
+
+/**
+ *  @brief  DiscreteProbabilityVector class
+ */
+class DiscreteProbabilityVector
+{
+public:
+
+    template <typename TX, typename TY>
```

Comment:
Have previously used typenames T and U, but TX and TY reads just fine.

Reply:
Yeah I think it helps with clarity if axis labels are used here to help convey what's going on.